### PR TITLE
Changed `MockPurchases` to use `PurchasesType`

### PR DESCRIPTION
**Pull request by NachoSoto**
Depends on https://github.com/RevenueCat/purchases-ios/pull/1912
This fixes the failures in https://github.com/RevenueCat/purchases-hybrid-common/pull/229

### CODE REVIEW

File: ios/PurchasesHybridCommon/PurchasesHybridCommonTests/Mocks/MockPurchases.swift [hunk 1]
```diff
@@ -7,147 +7,38 @@ import Foundation
...
-                   deviceCache: deviceCache,
-                   identityManager: identityManager,
-                   subscriberAttributes: subscriberAttributes,
-                   operationDispatcher: operationDispatcher,
-                   customerInfoManager: customerInfoManager,
-                   productsManager: productsManager,
-                   offeringsManager: offeringsManager,
-                   purchasesOrchestrator: purchasesOrchestrator,
-                   trialOrIntroPriceEligibilityChecker: trialOrIntroPriceEligibilityChecker)
-    }
```
Comment (NachoSoto) [inline ios/PurchasesHybridCommon/PurchasesHybridCommonTests/Mocks/MockPurchases.swift:123]: 👋🏻 🐼

File: ios/PurchasesHybridCommon/PurchasesHybridCommonTests/Mocks/MockPurchases.swift [hunk 2]
```diff
@@ -558,9 +462,135 @@ class MockPurchases: Purchases {
...
 
-    override func collectDeviceIdentifiers() {
+    func collectDeviceIdentifiers() {
         invokedCollectDeviceIdentifiers = true
         invokedCollectDeviceIdentifiersCount += 1
     }
     
 }
+
+extension MockPurchases {
```
Comment (NachoSoto) [inline ios/PurchasesHybridCommon/PurchasesHybridCommonTests/Mocks/MockPurchases.swift:472]: Now it becomes explicit that we're not overriding a bunch of methods, but this ensures that we don't use "the real" implementation when we don't want to.

File: ios/PurchasesHybridCommon/PurchasesHybridCommonTests/Mocks/MockPurchases.swift [hunk 3]
```diff
@@ -4,150 +4,41 @@
 //
 
 import Foundation
-@testable import RevenueCat
```
Comment (NachoSoto) [inline ios/PurchasesHybridCommon/PurchasesHybridCommonTests/Mocks/MockPurchases.swift:7]: No more need to use `internal` stuff :)

**Review by vegaro — APPROVED** [review]

**Review by tonidero — APPROVED** [review]
LGTM!

File: ios/PurchasesHybridCommon/PurchasesHybridCommonTests/Mocks/MockPurchases.swift [hunk 4]
```diff
@@ -208,7 +107,7 @@ class MockPurchases: Purchases {
     var invokedProductsParameters: (productIdentifiers: [String], completion: ([StoreProduct]) -> ())?
     var invokedProductsParametersList: [(productIdentifiers: [String], completion: ([StoreProduct]) -> ())] = []
 
-    override func getProducts(_ productIdentifiers: [String],
+    func getProducts(_ productIdentifiers: [String],
                               completion: @escaping ([StoreProduct]) -> ()) {
```
[the remark this answers is not in the document]
Comment (NachoSoto) [reply ios/PurchasesHybridCommon/PurchasesHybridCommonTests/Mocks/MockPurchases.swift:111]: Thanks, I did a find and replace and didn't fix these. We should add `SwiftLint` here too.

**Comment by NachoSoto** [comment]
We need to wait on https://github.com/RevenueCat/purchases-ios/pull/1912 to be released first.

**Comment by NachoSoto** [comment]
Integrated into #235 since we needed the version bump.

**Pull request closed without merging**